File: frontend/web/src/role-doctor/types/appointment.ts
import type { AppointmentNote } from "./appointmentNote"
import type { Schedule } from "./schedule"
import type { PatientInfo } from "./patient"

export interface Appointment {
  appointmentId: number
  patientId: number
  patientInfo: PatientInfo | null // Có thể null
  symptoms: string
  number: number
  schedule: Schedule
  appointmentStatus: "PENDING" | "CONFIRMED" | "COMPLETED" | "CANCELLED"
  createdAt: string
  appointmentNotes?: AppointmentNote | null
  slotEnd?: string // Format: "HH:mm:ss"
  slotStart?: string // Format: "HH:mm:ss"
}

// Interface cho response phân trang
export interface PaginatedResponse<T> {
  content: T[]
  pageNo: number
  pageSize: number
  totalElements: number
  totalPages: number
  last: boolean
}

// Interface cho filters frontend
export interface AppointmentFilters {
  status?: string
  date?: string
  scheduleId?: number
  searchTerm?: string
  gender?: string
}

export interface AppointmentStats {
  total: number
  pending: number
  confirmed: number
  completed: number
  cancelled: number
}

File: backend/appointment-service/src/main/java/org/example/appointmentservice/entity/Appointment.java
package org.example.appointmentservice.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.sql.Timestamp;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@Table(name = "appointments")
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "appointment_id")
    private Integer appointmentId;

    @Column(name = "doctor_id", nullable = false)
    private Integer doctorId;

    @Column(name = "patient_id", nullable = false)
    private Integer patientId;

    @Column(name = "schedule_id", nullable = false)
    private Integer scheduleId;

    @Column(columnDefinition = "TEXT")
    private String symptoms;

    private LocalTime slotStart;
    private LocalTime slotEnd;

    private Integer number;

    @Enumerated(EnumType.STRING)
    private AppointmentStatus appointmentStatus;

    @Column(name = "created_at")
    @CreationTimestamp
    private Timestamp createdAt;

    @OneToMany(mappedBy = "appointment", cascade = CascadeType.ALL)
    private List<AppointmentNote> appointmentNotes = new ArrayList<>();

    public enum AppointmentStatus {
        PENDING, CONFIRMED, CANCELLED, COMPLETED
    }
}
